File: SuperGrid/C#/ClipControl.cs
using System.Drawing;
using System.Windows.Forms;

namespace SuperGridDemo
{
    public partial class ClipControl : UserControl
    {
        public ClipControl()
        {
            InitializeComponent();
        }

        #region Public properties

        #region Text

        /// <summary>
        /// Gets or sets the control text
        /// </summary>
        public override string Text
        {
            get { return (base.Text); }

            set
            {
                labelX1.Text = value;

                base.Text = value;
            }
        }

        #endregion

        #region Font

        /// <summary>
        /// Gets or sets the control font
        /// </summary>
        public override Font Font
        {
            get { return (base.Font); }

            set
            {
                labelX1.Font = value;

                base.Font = value;
            }
        }

        #endregion

        #region ForeColor

        /// <summary>
        /// Gets or sets the control fore color
        /// </summary>
        public override Color ForeColor
        {
            get { return (base.ForeColor); }

            set
            {
                labelX1.ForeColor = value;

                base.ForeColor = value;
            }
        }

        #endregion

        #endregion
    }
}

File: SuperGrid/VB/ClipControl.vb
Namespace SuperGridDemo
	Partial Public Class ClipControl
		Inherits UserControl
		Public Sub New()
			InitializeComponent()
		End Sub

		#Region "Public properties"

		#Region "Text"

		''' <summary>
		''' Gets or sets the control text
		''' </summary>
		Public Overrides Property Text() As String
			Get
				Return (MyBase.Text)
			End Get

			Set(ByVal value As String)
				labelX1.Text = value

				MyBase.Text = value
			End Set
		End Property

		#End Region

		#Region "Font"

		''' <summary>
		''' Gets or sets the control font
		''' </summary>
		Public Overrides Property Font() As Font
			Get
				Return (MyBase.Font)
			End Get

			Set(ByVal value As Font)
				labelX1.Font = value

				MyBase.Font = value
			End Set
		End Property

		#End Region

		#Region "ForeColor"

		''' <summary>
		''' Gets or sets the control fore color
		''' </summary>
		Public Overrides Property ForeColor() As Color
			Get
				Return (MyBase.ForeColor)
			End Get

			Set(ByVal value As Color)
				labelX1.ForeColor = value

				MyBase.ForeColor = value
			End Set
		End Property

		#End Region

		#End Region
	End Class
End Namespace
